Mapbox/MapboxMapsStyle/Types/Color.swift

```diff
@@ -1,48 +1,34 @@
 import UIKit
 
-public typealias ColorRepresentable = String
-extension ColorRepresentable {
+/// Container to represent `UIColor`for use by the map renderer
+public struct ColorRepresentable: Codable, Equatable {
+
+    /// String representation of a `UIColor` used by the renderer
+    public let colorRepresentation: String?
 
     /// Create a string representation of a `UIColor`
-    public init(color: UIColor) {
+    /// - Parameter color: A `UIColor` instance in the sRGB color space
+    /// - Returns: Initializes a `ColorRepresentable` instance if the `color` is in sRGB color space. Returns `nil` otherwise.
+    public init?(color: UIColor) {
```

Comment:
I'm not convinced that making this initializer failable results in a better API. In the modified examples above, we now have to force unwrap. My guess is that most consumers would do the same, at which point we might as well just have a fatal error on line 20 below instead of returning nil.

Reply:
Yeah, I'm not opposed to that.. the API does feel sub-optimal. The only thing is we don't seem to be using `fatalError` anywhere else in the SDK.

Reply:
Let's use `fatalError` and see where else it may be a better fit in the sdk.